File: apps/web/lib/schemas/pronunciation.ts
import { z } from "zod";

import { PronunciationSourceSchema } from "@/lib/schemas/text";

export const PronunciationCandidateSchema = z.object({
  phonemes: z.string(),
  confidence: z.number(),
  source: PronunciationSourceSchema,
});
export type PronunciationCandidate = z.infer<typeof PronunciationCandidateSchema>;

export const PronunciationCandidatesResponseSchema = z.object({
  surface: z.string(),
  candidates: z.array(PronunciationCandidateSchema),
});
export type PronunciationCandidatesResponse = z.infer<typeof PronunciationCandidatesResponseSchema>;

export const PronunciationPreviewResponseSchema = z.object({
  sample_rate: z.number(),
  channels: z.number(),
  bit_depth: z.number(),
  pcm_base64: z.string(),
});
export type PronunciationPreviewResponse = z.infer<typeof PronunciationPreviewResponseSchema>;

File: apps/api/app/schemas/pronunciation.py
from __future__ import annotations

from pydantic import BaseModel


class PronunciationCandidateSchema(BaseModel):
    phonemes: str
    confidence: float
    source: str


class PronunciationCandidatesRequest(BaseModel):
    surface: str
    language: str
    script: str


class PronunciationCandidatesResponse(BaseModel):
    surface: str
    candidates: list[PronunciationCandidateSchema]


class PronunciationPreviewRequest(BaseModel):
    surface: str
    phonemes: str
    voice_id: str


class PronunciationPreviewResponse(BaseModel):
    sample_rate: int
    channels: int
    bit_depth: int
    pcm_base64: str
